### PR TITLE
	RavenDB-7440 and RavenDB-6716 fixes

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Client/Documents/Commands/Batches/BatchCommand.cs [hunk 1]
```diff
@@ -37,7 +37,12 @@ public BatchCommand(DocumentConventions conventions, JsonOperationContext contex
                 {
                     if (_attachmentStreams == null)
```
Comment (ayende) [inline src/Raven.Client/Documents/Commands/Batches/BatchCommand.cs:38]: Change this to a HashSet, instead.

File: src/Raven.Client/Documents/Commands/Batches/BatchCommand.cs [hunk 2]
```diff
@@ -37,7 +37,12 @@ public BatchCommand(DocumentConventions conventions, JsonOperationContext contex
...
                         _attachmentStreams = new List<Stream>();
-                    _attachmentStreams.Add(putAttachmentCommandData.Stream);
+
+                    var stream = putAttachmentCommandData.Stream;
+                    PutAttachmentCommandHelper.ValidateStream(stream);
+                    if (_attachmentStreams.Contains(stream))
```
Comment (ayende) [inline src/Raven.Client/Documents/Commands/Batches/BatchCommand.cs:43]: Do the check in the Add

Comment (fitzchak) [reply src/Raven.Client/Documents/Commands/Batches/BatchCommand.cs:43]: Done

File: src/Raven.Client/Documents/Commands/Batches/BatchCommand.cs (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Client/Documents/Commands/Batches/BatchCommand.cs:38]: Done

**Pull request merged**